File: app/src/main/java/com/jbrunton/mymovies/api/resources/BaseMovieResource.java
package com.jbrunton.mymovies.api.resources;

import com.google.common.base.Optional;
import com.jbrunton.entities.Configuration;
import com.jbrunton.entities.Movie;

import org.joda.time.LocalDate;

public class BaseMovieResource {
    private String id;
    private String originalTitle;
    private String posterPath;
    private String backdropPath;
    private LocalDate releaseDate;
    private String voteAverage;

    protected Movie.Builder builder(Configuration config) {
        return Movie.builder()
                .id(id)
                .title(originalTitle)
                .posterUrl(config.expandUrl(posterPath))
                .backdropUrl(config.expandUrl(backdropPath))
                .releaseDate(Optional.fromNullable(releaseDate))
                .rating(voteAverage);
    }
}

File: networking/src/main/java/com/jbrunton/networking/resources/movies/BaseMovieResource.kt
package com.jbrunton.networking.resources.movies

import org.joda.time.LocalDate

interface BaseMovieResource {
    val id: String
    val originalTitle: String
    val posterPath: String
    val backdropPath: String
    val releaseDate: LocalDate?
    val voteAverage: String

//    protected fun builder(config: Configuration): Movie.Builder {
//        return Movie.Companion.builder()
//                .id(id)
//                .title(originalTitle)
//                .posterUrl(config.expandUrl(posterPath))
//                .backdropUrl(config.expandUrl(backdropPath))
//                .releaseDate(Optional.fromNullable<T>(releaseDate))
//                .rating(voteAverage)
//    }
}
